src/io/flutter/sdk/FlutterSdk.java

```diff
@@ -32,13 +38,21 @@
 import io.flutter.console.FlutterConsoleHelper;
 import io.flutter.dart.DartPlugin;
 import io.flutter.utils.FlutterModuleUtils;
+import org.jdom.Element;
+import org.jdom.JDOMException;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
+import java.io.File;
+import java.io.IOException;
 import java.util.concurrent.atomic.AtomicBoolean;
 
 public class FlutterSdk {
   public static final String FLUTTER_SDK_GLOBAL_LIB_NAME = "Flutter SDK";
+
+  private static final String SDK_LIBRARY_PATH = ".idea/libraries/Dart_SDK.xml";
```

Comment:
You may consider using a `StartupActivity` instead of `ProjectOpenProcessor` to recover the SDK. Inside a `StartupActivity` you have an instance of `Project` and can use all the Platform API mthods, so there will be no suspicious string literals in code or hacky `loadDartSdkLibrary()` method.